framework/test/unit/modules/interoperability/mainchain/commands/message_recovery.spec.ts

```diff
@@ -436,119 +590,40 @@ describe('Mainchain MessageRecoveryCommand', () => {
 			terminatedOutboxAccountMock.has.mockResolvedValue(true);
 		});
 
-		it('should successfully process recovery transaction', async () => {
-			// Act
-			chainAccountStoreMock.has.mockResolvedValue(true);
-			await command.execute(commandExecuteContext);
-			expect.assertions(ccms.length + 1);
-
-			{
-				// Arrange
-				const { chainID } = transactionParams;
-				const outboxRoot = Buffer.alloc(32);
-
-				// Assert
-				expect(terminatedOutboxAccountMock.set).toHaveBeenCalledWith(
-					expect.anything(),
-					chainID,
-					expect.objectContaining({
-						outboxRoot,
-					}),
-				);
-			}
-
-			for (const ccm of ccms) {
-				// Assign
-				const chainID = ccm.sendingChainID;
-				// Assert
-				expect(command['internalMethod'].addToOutbox).toHaveBeenCalledWith(
-					expect.anything(),
-					chainID,
-					swapReceivingAndSendingChainIDs(ccm),
-				);
-			}
-		});
-
-		it('should throw when beforeRecoverCCM of ccMethods of the ccm fails', async () => {
-			// Assign & Arrange
-			const method = ({
-				beforeRecoverCCM: jest.fn(() => {
-					throw new Error('beforeRecoverCCM Error');
-				}),
-				MODULE_NAME_INTEROPERABILITY,
-			} as unknown) as BaseCCMethod;
-
-			command['interoperableCCMethods'].set(MODULE_NAME_INTEROPERABILITY, method);
-
-			// Assert
-			await expect(command.execute(commandExecuteContext)).rejects.toThrow(
-				'beforeRecoverCCM Error',
-			);
-		});
-
-		it('should throw when terminated chain outbox does not exist', async () => {
-			// Assign & Arrange
-			const { chainID } = transactionParams;
-
-			when(terminatedOutboxAccountMock.has).calledWith(chainID).mockResolvedValue(false);
-
-			// Assert
-			await expect(command.execute(commandExecuteContext)).rejects.toThrow(
-				'Terminated outbox account does not exist',
-			);
-		});
-
-		it('should not add CCM to outbox when sending chain of the CCM does not exist', async () => {
-			// Assign & Arrange & Act
-			chainAccountStoreMock.has.mockResolvedValue(false);
-
-			await command.execute(commandExecuteContext);
-
-			// Assert
-			expect.assertions(ccms.length);
-			for (const _ of ccms) {
-				expect(command['internalMethod'].addToOutbox).not.toHaveBeenCalled();
-			}
-		});
-
-		it('should not add CCM to outbox when sending chain of the CCM is not live', async () => {
-			// Assign & Arrange & Act
-			for (const ccm of ccms) {
-				const chainID = ccm.sendingChainID;
-
-				when(command['internalMethod'].isLive as never)
-					.calledWith(chainID)
-					.mockResolvedValue(false as never);
-			}
-
-			await command.execute(commandExecuteContext);
+		it('should proceed without error if the sending chain is the mainchain', async () => {
```

Comment:
should call applyRecovery ? and also check applyRecovery to have been called with right params?